src/main/groovy/util/StockMovementStatusHelper.groovy

```diff
@@ -69,15 +31,78 @@ class StockMovementStatusHelper {
+        return shipment?.status?.code ?: ShipmentStatusCode.PENDING
+    }
+
+    static Enum getStatus(StockMovementContext context) {
+        if (context.isReturn()) {
+            return context.isInbound()
+                    ? getInboundReturnDisplayStatus(context.shipment)
+                    : getOutboundReturnDisplayStatus(context.order)
+        }
+
+        if (context.isFromPurchaseOrder()) {
+            return context?.shipment?.status?.code
         }
-        return defaultStatus
+
+        if (context.requisition?.status == RequisitionStatus.ISSUED && context?.shipment?.hasShipped()) {
+            return context?.shipment?.status?.code
+        }
+
+        return context.requisition?.status
     }
-}
 
+    static Enum getListStatus(StockMovementContext context) {
+        if (context.isReturn()) {
+            return context.isInbound()
+                    ? getInboundReturnDisplayStatus(context.shipment)
+                    : getOutboundReturnDisplayStatus(context.order)
+        }
+
+        if (context.isInbound()) {
+            //Request
+            if (context.isElectronicType() && context.isCurrentLocationDownstreamConsumer()) {
+                // Approval request
+                // Mapping statuses for display for the requestor's dashboard
+                // We want to display all statuses from approval workflow (approved, rejected and pending approval)
+                if (context?.requisition?.status in RequisitionStatus.listApproval()) {
+                    switch(context?.requisition?.status) {
+                        case RequisitionStatus.APPROVED:
+                            return StockMovementStatusCode.APPROVED
+                        case RequisitionStatus.REJECTED:
+                            return StockMovementStatusCode.REJECTED
+                        case RequisitionStatus.PENDING_APPROVAL:
+                            return StockMovementStatusCode.PENDING_APPROVAL
+                    }
+                }
+
+                // Regular request
+                // We want to map all statuses from depot side to "in progress".
+                if (context?.requisition?.status in RequisitionStatus.listRequestOptions()) {
+                    return StockMovementStatusCode.IN_PROGRESS
+                }
+            }
+
+            return context?.shipment?.status?.code
+        }
+
+        if (context.isOutbound()) {
+            return context?.requisition?.status
+        }
+
+        return context?.shipment?.status?.code ?: context?.requisition?.status
+    }
+
+    static def getApplicationTagLib() {
+        return Holders.grailsApplication.mainContext.getBean('org.grails.plugins.web.taglib.ApplicationTagLib')
+    }
 
+    static Map getStatusMetaData(Enum status) {
+        return [
+                name: status?.name(),
+                label: applicationTagLib.message(code: 'enum.' + status?.getClass()?.getSimpleName() + '.' + status),
```

Comment:
shouldn't we add a default message in case something does not have the translated label?